### PR TITLE
Fix some build problem in linux

### PR DESCRIPTION
**Pull request by jxo**
fix some build problem in linux
1. project treats warning as error, we should add more -Wno-error for linux
2. some stl header files should be included in linux
3. some switch clause passthrough fix

### CODE REVIEW

File: src/Core/MapUtil/RmgUtil/TemplateUtils.hpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ constexpr inline int64_t intSqrt(const int64_t value) noexcept
     int64_t result = 3;
     if (!std::is_constant_evaluated()) { // DO NOT 'if constexpr' here! Also, replace to 'if consteval' later.
         // speedup so runtime will only use 1-2 loop iterations.
-        result = static_cast<int64_t>(std::sqrtl(value)) - 1;
+        result = static_cast<int64_t>(sqrtl(value)) - 1;
```
Comment (mapron) [inline src/Core/MapUtil/RmgUtil/TemplateUtils.hpp:133]: include <cmath> instead? that must be in std:: namespace.

File: CMakeLists.txt [hunk 2]
```diff
@@ -258,6 +266,7 @@ AddTarget(TYPE shared NAME GuiUtils OUTPUT_PREFIX FH
 AddTarget(TYPE shared NAME GuiWidgets OUTPUT_PREFIX FH
     SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/Gui/Widgets
     EXPORT_INCLUDES
+    INCLUDES ${CMAKE_CURRENT_SOURCE_DIR}/src/Gui/Widgets
```
Comment (mapron) [inline CMakeLists.txt:269]: No, that should not be that way, I'd better fix AddTarget

Comment (mapron) [reply CMakeLists.txt:269]: I added a fix to the mernel, can you rebase to the master branch and remove extra INCLUDES here?

**Review by mapron — CHANGES_REQUESTED** [review]
Couple comments. otherwise good.

File: CMakeLists.txt (hunk 2 above)
Comment (jxo) [reply CMakeLists.txt:269]: Yes, you are right

File: src/Core/MapUtil/RmgUtil/TemplateUtils.hpp (hunk 1 above)
Comment (jxo) [reply src/Core/MapUtil/RmgUtil/TemplateUtils.hpp:133]: I find that sqrtl not in std in cmath in linux, but i don't know waht

Comment (mapron) [reply src/Core/MapUtil/RmgUtil/TemplateUtils.hpp:133]: Well, okay... leave it as is

**Comment by jxo** [comment]
> Couple comments. otherwise good.

Is that ok?

**Comment by mapron** [comment]
> Is that ok?

Well, what? Sorry?

I am waiting for you to rebase onto current master and remove unnessesary INCLUDE, that's it. What's unclear?

**Review by mapron — APPROVED** [review]

**Pull request merged**